e2e/stack_test.go

```diff
@@ -317,52 +186,70 @@ func ethRollupFlow(t *testing.T, stack *e2e.StackConfig) {
 
 	t.Log("Monomer can ingest user deposit txs from L1 and mint ETH on L2")
 
 	/////////////////////////////
 	////// ETH WITHDRAWALS //////
 	/////////////////////////////
 
+	withdrawalAmount := big.NewInt(1)
+
 	// create a withdrawal tx to withdraw the deposited amount from L2 back to L1
-	withdrawalTx := e2e.NewWithdrawalTx(0, common.Address(userCosmosETHAddress), userETHAddress, depositAmount, new(big.Int).SetUint64(params.TxGas))
+	withdrawalTx := e2e.NewWithdrawalTx(0, common.Address(userCosmosETHAddress), userETHAddress, withdrawalAmount, new(big.Int).SetUint64(params.TxGas))
 
-	// initiate the withdrawal of the deposited amount on L2
-	senderAddr, err := monomer.CosmosETHAddress(*withdrawalTx.Sender).Encode("cosmos")
+	queryAccountBytes, err := protov1.Marshal(&authv1beta1.QueryAccountRequest{
+		Address: userCosmosAddr,
+	})
+	require.NoError(t, err)
+	result, err := stack.L2Client.ABCIQuery(stack.Ctx, authv1beta1.Query_Account_FullMethodName, queryAccountBytes)
+	require.NoError(t, err)
+	require.Zero(t, result.Response.Code)
+	var accountResponse authv1beta1.QueryAccountResponse
+	require.NoError(t, protov1.Unmarshal(result.Response.Value, &accountResponse))
+	var baseAccount authv1beta1.BaseAccount
+	require.NoError(t, protov1.Unmarshal(accountResponse.Account.Value, &baseAccount))
+
+	l2ChainID, err := stack.MonomerClient.ChainID(stack.Ctx)
 	require.NoError(t, err)
+
 	withdrawalTxResult, err := stack.L2Client.BroadcastTxAsync(
 		stack.Ctx,
-		testapp.ToTx(t, &rolluptypes.MsgInitiateWithdrawal{
-			Sender:   senderAddr,
-			Target:   withdrawalTx.Target.String(),
-			Value:    math.NewIntFromBigInt(withdrawalTx.Value),
-			GasLimit: withdrawalTx.GasLimit.Bytes(),
-			Data:     []byte{},
+		buildTx(t, l2ChainID.String(), baseAccount.Sequence, baseAccount.AccountNumber, userPrivKey, []protov1.Message{
+			&rolluptypes.MsgInitiateWithdrawal{
+				Sender:   userCosmosAddr,
+				Target:   withdrawalTx.Target.String(),
+				Value:    math.NewIntFromBigInt(withdrawalTx.Value),
+				GasLimit: withdrawalTx.GasLimit.Bytes(),
+				Data:     []byte{},
+			},
 		}),
 	)
 	require.NoError(t, err)
-	require.Equal(t, abcitypes.CodeTypeOK, withdrawalTxResult.Code)
+	require.Equalf(t, abcitypes.CodeTypeOK, withdrawalTxResult.Code, "log: "+withdrawalTxResult.Log)
 
 	// wait for tx to be processed on L2
-	require.NoError(t, stack.WaitL2(1))
+	require.NoError(t, stack.WaitL2(2))
 
 	// inspect L2 events to ensure that the user's ETH was burned on L2
-	requireEthIsBurned(t, stack, userCosmosAddr, depositValueHex)
+	requireEthIsBurned(t, stack, userCosmosAddr, hexutil.EncodeBig(withdrawalAmount))
 
 	// wait for the L2 output containing the withdrawal tx to be proposed on L1
-	l2OutputBlockNumber := waitForL2OutputProposal(t, stack.L2OutputOracleCaller)
+	var l2OutputBlockNumber *big.Int
+	for range 5 {
+		l2OutputBlockNumber = waitForL2OutputProposal(t, stack.L2OutputOracleCaller)
+	}
```

Comment:
Why do we need to wait for five output proposals?

Reply:
Waiting for one wasn't enough. I don't understand the root cause. Would be good to understand this.